src/cxx_support.rs

```diff
@@ -0,0 +1,214 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Support for cxx types.
+
+use std::pin::Pin;
+
+use cxx::{memory::UniquePtrTarget, UniquePtr};
+
+use crate::{Emplace, TryNew};
+
+/// Equivalent of a [`UniquePtr`] which is allocated, but hasn't yet
+/// had its constructor called.
+#[cfg(feature = "cxx")]
+pub struct MaybeUninitUniquePtr<T: UniquePtrTarget>(*mut T);
```

Comment:
I'm not sure I understand what the value of this newtype is. Would it ever appear in user code beyond at the boundaries of MakeUninitUniquePtr? Feels like that's just a glorified bridge over top of operator new/operator delete, which work with void*s.

Reply:
I'm going to try getting rid of it. I think I agree with you.